persistence/src/commonMain/db_user/com/wire/kalium/persistence/Messages.sq

```diff
@@ -349,10 +355,14 @@ SELECT (
 
 getLastMessages:
 SELECT * FROM MessagePreview AS message
-WHERE visibility = 'VISIBLE' AND contentType IN ('TEXT', 'ASSET', 'KNOCK', 'MISSED_CALL', 'CONVERSATION_RENAMED', 'MEMBER_CHANGE')
-GROUP BY conversationId
-HAVING date = MAX(date)
-ORDER BY message.date ASC;
+WHERE id IN (
```

Comment:
Nested queries are expensive than the joins, if we can convert that to a join that would be great!

Reply:
I don't see a way to use a join for this 🤔 any tips?


Also, for what it's worth, there are no sub trees in the `EXPLAIN QUERY PLAN` for this one.